%% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax
\ifdefined\pdfimageresolution
    \pdfimageresolution= \numexpr \dimexpr1in\relax/\sphinxpxdimen\relax
\fi
%% let collapsible pdf bookmarks panel have high depth per default
\PassOptionsToPackage{bookmarksdepth=5}{hyperref}


\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
  \ifdefined\DeclareUnicodeCharacterAsOptional
    \def\sphinxDUC#1{\DeclareUnicodeCharacter{"#1}}
  \else
    \let\sphinxDUC\DeclareUnicodeCharacter
  \fi
  \sphinxDUC{00A0}{\nobreakspace}
  \sphinxDUC{2500}{\sphinxunichar{2500}}
  \sphinxDUC{2502}{\sphinxunichar{2502}}
  \sphinxDUC{2514}{\sphinxunichar{2514}}
  \sphinxDUC{251C}{\sphinxunichar{251C}}
  \sphinxDUC{2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}



\usepackage{tgtermes}
\usepackage{tgheros}
\renewcommand{\ttdefault}{txtt}



\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=auto}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}


\usepackage{sphinxmessages}




\title{ngsSRA}
\date{Dec 20, 2022}
\release{0.1}
\author{Pavlos Pavlidis}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Release}
\makeindex
\begin{document}

\ifdefined\shorthandoff
  \ifnum\catcode`\=\string=\active\shorthandoff{=}\fi
  \ifnum\catcode`\"=\active\shorthandoff{"}\fi
\fi

\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}


\sphinxAtStartPar
\sphinxstylestrong{The NGS tutorial} will guide you from raw SRA files to VCF variants.
We start with the SRA data from NCBI. Then, we convert them to fastq using the sra\sphinxhyphen{}toolkit and then we apply samtools and bcftools to obtain variants.
We will use A. thaliana data to demonstrate the pipeline


\chapter{1.1   Download some data}
\label{\detokenize{index:download-some-data}}
\sphinxAtStartPar
Perhaps, this is the most difficult part of the whole pipeline i.e., \sphinxstylestrong{where to find data}. Usually, we download data from the NCBI, ENA or other resources, such as the 1000Genomes project or the Arabidopsis project or other web databases.


\chapter{1.2   Overview}
\label{\detokenize{index:overview}}
\sphinxAtStartPar
The following steps will be followed:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Download and install the \sphinxhref{https://github.com/ncbi/sra-tools}{sra\sphinxhyphen{}toolkit}.

\item {} 
\sphinxAtStartPar
Get some data from NCBI (it is supposed that you already know the ID of the runs)

\item {} 
\sphinxAtStartPar
Download the SRA data

\item {} 
\sphinxAtStartPar
Convert the data to fastq format

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{Mapping}: Run \sphinxtitleref{bwa mem} to perform mapping

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{SAMTOOLS}: Samtools will do some additional operations, such a sorting, fixing the mate flags

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{Calling}: Use bcftools to call the genotypes and produce the GVCF files.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{Merge the files}: Merge all individual files to a single file containing the information from all sequencded individuals.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{Final SNP calling and VCF production}

\sphinxAtStartPar
The final product will be a single VCF file that contains \sphinxstylestrong{just the SNPs}. Here, there is a specific definition of what a SNP is: \sphinxstylestrong{A SNP is a position that at least one individual is different from the reference}. This means, that \sphinxstylestrong{a SNP might actually be monomorphic if all individuals are homozygous for the state that is not present in the reference genome}.

\end{enumerate}


\chapter{1.3   Download and install the sra\sphinxhyphen{}toolkit}
\label{\detokenize{index:download-and-install-the-sra-toolkit}}\begin{quote}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Download and install \sphinxhref{https://github.com/ncbi/sra-tools}{sra\sphinxhyphen{}toolkit}.
Even if you can build the code from the source, luckily, there is a link to pre\sphinxhyphen{}compiled software available for you system. The link is here \sphinxhref{https://github.com/ncbi/sra-tools/wiki/01.-Downloading-SRA-Toolkit}{PRECOMPILED SRA\sphinxhyphen{}TOOLKIT}. For my system, I used the \sphinxhref{https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/3.0.2/sratoolkit.3.0.2-ubuntu64.tar.gz}{Ubuntu Linux 64bit architecture}, even though I’m running a Debian system. It works fine.

\item {} 
\sphinxAtStartPar
After you download and unzip the code, you can navigate inside the \sphinxstylestrong{sratoolkit.3.0.1\sphinxhyphen{}ubuntu64} folder and then inside the \sphinxstylestrong{bin} directory. There, you can see the following files:

\end{enumerate}

\noindent\sphinxincludegraphics[width=800\sphinxpxdimen]{{sradir}.png}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
It is comfortable to modify your PATH variable, to include this specific directory in the PATH variable.
* Open the \sphinxtitleref{\textasciitilde{}/.bashrc} file (e.g. \sphinxcode{\sphinxupquote{emacs \sphinxhyphen{}nw \textasciitilde{}/.bashrc}} or \sphinxcode{\sphinxupquote{vim \textasciitilde{}/.bashrc}} or \sphinxcode{\sphinxupquote{nano \textasciitilde{}/.bashrc}}) and go to the end of the file. Then type something like:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n+nv}{PATH}\PYG{o}{=}/home/pavlos/software/sratoolkit.3.0.1\PYGZhy{}ubuntu64/bin/:\PYG{n+nv}{\PYGZdl{}PATH}
\PYG{n+nb}{export} PATH
\end{sphinxVerbatim}

\sphinxAtStartPar
In this case, the \sphinxcode{\sphinxupquote{/home/pavlos/software/sratoolkit.3.0.1\sphinxhyphen{}ubuntu64/bin/}} is the directory that I want to add in the PATH. Please MODIFY it according to your own path. Then, save and close the file and type in the terminal:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n+nb}{source} \PYGZti{}/.bashrc  \PYG{c+c1}{\PYGZsh{}\PYGZsh{} or . \PYGZti{}/.bashrc}
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
Do NOT forget the \$PATH part at the end of the PATH=…. Otherwise, you will destroy the PATH variable of the system. Thus, you actually append to the PATH variable the directory that you need.
\end{sphinxadmonition}

\end{enumerate}
\end{quote}


\chapter{1.4   Download some data using the sra\sphinxhyphen{}toolkit}
\label{\detokenize{index:download-some-data-using-the-sra-toolkit}}
\sphinxAtStartPar
Now, that all the tools from the sra\sphinxhyphen{}toolkit software package, we are ready to download some data from the SRA database of the NCBI. The SRA file format contains all the information (sequence and base qualities) but they cannot be used directly for the downstream analyses. Thus, we will download them and then we will convert them to FASTQ files.

\sphinxAtStartPar
Finding the proper SRA files isn’t a very trivial task, since \sphinxstylestrong{there are tones of data out there… but they are not so well organized}. For example, at least, for me it’s not trivial at all how to get all SRA files for humans that sampled in Europe. Anyway, in this tutorial we will cheat a bit and we will give you just 3 SRA IDs for 3 Arabidopsis plants. If we have a list of SRA files, then it’s easy to get the data. The difficult task is to find the SRA IDs we need.

\sphinxAtStartPar
The data we will download have the following SRA IDs:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
SRR1945435
SRR1945436
SRR1945437
\end{sphinxVerbatim}

\sphinxAtStartPar
To get a file, we will use the \sphinxcode{\sphinxupquote{prefetch}} command from the sra\sphinxhyphen{}toolkit. Thus just type:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
prefetch SRR1945435
\end{sphinxVerbatim}

\sphinxAtStartPar
This will just download the SRR1945435 file. If you have saved all the IDs in a file called e.g., “accesions.txt”, you can type

\begin{sphinxVerbatim}[commandchars=\\\{\}]
prefetch \PYGZhy{}\PYGZhy{}option\PYGZhy{}file accesions.txt
\end{sphinxVerbatim}

\sphinxAtStartPar
This will take \sphinxstylestrong{some time}. The SRA files are quite large, so please be patient. Also, you need to have \sphinxstylestrong{sufficient amount of avaialbe hard disk space}.


\chapter{1.5   Convert the data from SRA to fastq}
\label{\detokenize{index:convert-the-data-from-sra-to-fastq}}
\sphinxAtStartPar
We will use the \sphinxcode{\sphinxupquote{fasterq\sphinxhyphen{}dump}} to convert the downloaded SRA files to the fastq format. Here we assume paired\sphinxhyphen{}end format, so two files will be generated.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
If our reads are paired\sphinxhyphen{}end, then two fastq files will be created. The first will look like *\_1.fastq.gz and the second *\_2.fastq.gz.
\end{sphinxadmonition}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
fasterq\PYGZhy{}dump SRR1945437.sra
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The sra files have been stored within folders entitled as their ID. Thus a more efficient way to convert all the SRA to fastq is something like the following command:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k}{for} file \PYG{k}{in} \PYG{l+s+sb}{`}find \PYGZhy{}iname \PYG{l+s+s1}{\PYGZsq{}*.sra\PYGZsq{}} \PYG{p}{|} xargs ls\PYG{l+s+sb}{`} \PYG{c+c1}{\PYGZsh{}\PYGZsh{} The \PYGZdq{}| xargs ls\PYGZdq{} part can be removed. I just want to have the files alphabetically ordered.}
\PYG{k}{do}
\PYG{n+nb}{echo} \PYG{n+nv}{\PYGZdl{}file}\PYG{p}{;}
fasterq\PYGZhy{}dump \PYG{n+nv}{\PYGZdl{}file}\PYG{p}{;}
\PYG{k}{done}
\end{sphinxVerbatim}
\end{sphinxadmonition}

\sphinxAtStartPar
The fastq files will be stored in the folder that we run the command.


\chapter{1.6   The Quality Control and the mapping process}
\label{\detokenize{index:the-quality-control-and-the-mapping-process}}
\sphinxAtStartPar
The quality control (QC) is an important part of the analysis since it can save you from the analysis of low\sphinxhyphen{}quality data. We will go through that in another tutorial. For now, let’s suppose that the data is of good quality and proceed with the mappign process
\begin{quote}
\end{quote}



\renewcommand{\indexname}{Index}
\printindex
\end{document}